__tests__/ConnectionScreen-test.tsx

```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+import { screen, render, fireEvent } from '@testing-library/react-native';
+import ConnectionScreen from '../src/screens/ConnectionScreen/ConnectionScreen';
+
+describe('ConnectionScreen component', () => {
+  it('renders properly and triggers retry function on button press', () => {
+    const retryConnectMock = jest.fn();
+    render(<ConnectionScreen retryConnect={retryConnectMock} />);
+    expect(screen.getByText('Retry')).toBeTruthy();
+    fireEvent.press(screen.getByText('Retry'));
+    expect(retryConnectMock).toHaveBeenCalledTimes(1);
+  });
+});
```

Comment:
Can we have tests for screen when internet is connected, and when internet is not connected, and when internet is back

Reply:
Yes we can. I'm creating an issue ticket to fix the test in this project after completion of moving this to lts version. Can I do that in that ticket?